controllers/replication/secret_controller.go

```diff
@@ -455,23 +460,30 @@ func getPrefixedSecretName(prefix string, secretName string) string {
 	return fmt.Sprintf("%s%s", prefix, secretName)
 }
 
-func contains(s string, arr []string) bool {
-	for _, i := range arr {
-		if i == s {
-			return true
-		}
-	}
-	return false
-}
-
 func calculateTargetLabels(originalLabels map[string]string, target api.ReplicationTarget) map[string]string {
+	out := make(map[string]string)
+	for k, v := range originalLabels {
+		out[k] = v
+	}
 	for k, v := range target.AddLabels {
-		originalLabels[k] = v
+		out[k] = v
+	}
+	for _, key := range target.DropLabels {
+		delete(out, key)
 	}
-	for key := range originalLabels {
-		if contains(key, target.DropLabels) {
-			delete(originalLabels, key)
+	return out
+}
+
+func wouldBeInfinite(origin corev1.Secret, rsec api.ReplicatedSecret, target api.ReplicationTarget) bool {
+	computedLabels := calculateTargetLabels(origin.Labels, target)
+	for k, v := range rsec.Spec.Selector.MatchLabels {
+		if computedLabels[k] != v {
+			return false
 		}
 	}
```

Comment:
[nitpick] Although I think this is the correct selector logic, I'd rather not reimplement it in our code. Instead of `rsec`, we can pass the `selector` variable that is already computed in the calling method, and then rewrite this as:
```go
if !selector.Matches(labels.Set(computedLabels)) {
  return false
}
```

Reply:
I'd rather avoid this. Delegating to a function here just means you have to context switch more, and this is literally 2 lines of logic.

Reply:
[issue] Actually this is not the correct logic (I missed that in my first review).
A ReplicatedSecret can also use `Selector.MatchExpressions` with operators (`In`, `NotIn`, etc).
`Matches()` will handle that.

Reply:
Actually I think you're right, if I look here, the selector is indeed used inside a list call [here](https://github.com/k8ssandra/k8ssandra-operator/blob/feature/finite-replicatedsecrets/controllers/replication/secret_controller.go#L383-L386), so I think there is an argument that `.Matches` indeed will allow the expanded functionality of those operators. I'm not sure if that's used anywhere, but we may as well support it as it is a part of the embedded k8s API.

Thanks for identifying this, I'll make the change.